sql_file3.sql

```diff
@@ -0,0 +1,705 @@
+
+-- 9 Compare the results of character count and byte count for the string 'SQL Server' with a trailing space.
+
+SELECT LEN('SQL Server') AS StringLength, DATALENGTH('SQL Server') AS ByteLength; 
+
+
+-- 10 Write a query to find the byte count of an empty string and explain the result.
+
+SELECT LEN('')  AS ByteLength; 
+
+-- 11 Find the position of 'Server' in the string 'Microsoft SQL Server'.
+
+SELECT CHARINDEX('Server','Microsoft SQL Server') AS ServerPosition;
+
+
+
+-- 12 From an 'Emails' table, write a query to extract the domain name from email addresses.
+
+CREATE TABLE Emails(
+EmailId INT PRIMARY KEY,
+Email VARCHAR(100));
+
+INSERT INTO Emails VALUES
+(1,'mukhil.mp@gmail.com'),
+(2,'mukhil.mp@geekywolf.com'),
+(3,'mukhil.mp@Yahoo.com');
+
+SELECT Email, RIGHT(Email,LEN(Email)-CHARINDEX('@',Email)) as Domian 
+FROM Emails;
+
+-- 13 Find the position of the first number in the string 'ABC123DEF456'.
+
+SELECT PATINDEX('%[0-9]%','ABC123DEF456') AS FirstNumber;
+
+--14 Write a query to find all product names from a 'Products' table that contain a number.
+
+select * from products;
+
+ALTER TABLE Products
+ADD ProductName VARCHAR(50);
+
+UPDATE Products
+SET ProductName = CASE 
+    WHEN ProductID = 'A1YZT' THEN 'APLLE123'
+    WHEN ProductID = 'AK47S' THEN 'ORANGE'
+    WHEN ProductID = 'B2ORH' THEN 'GRA456PS'
+    ELSE ProductName
+END
+WHERE ProductID IN ('A1YZT','AK47S','B2ORH');
+
+SELECT ProductName
+FROM Products
+WHERE ProductName LIKE '%[0-9]%';
+
+-- 15 Join the strings 'SQL', 'Server', and '2022' with spaces between them
+
+SELECT 'SQL'+' '+'Server'+' '+'2022' AS CombinedString;
+
+-- 16 From 'Employees' table with 'FirstName' and 'LastName' columns, create a 'FullName' column.
+
+USE TASK2;
+SELECT name AS TableName 
+FROM sys.tables;
+
+SELECT * FROM Employees;
+
+SELECT (FirstName+' '+LastName) AS FullName 
+FROM Employees;
+
+-- 17 Join the array ('SQL', 'Server', '2022') with a hyphen as the separator.
+USE TASK3;
+
+SELECT CONCAT('SQL','-', 'Server','-', '2022') AS String;
+
+-- 18  From an 'Addresses' table, combine 'Street', 'City', 'State', and 'ZIP' columns into a single address string
+
+CREATE TABLE Address(
+Street VARCHAR(50),
+City VARCHAR(50),
+State VARCHAR(50),
+ZIP INT);
+
+INSERT INTO Address (Street, City, State, ZIP) VALUES
+('123 Main St', 'Springfield', 'IL', 62701),
+('456 Elm St', 'Chicago', 'IL', 60601),
+('789 Maple Ave', 'Los Angeles', 'CA', 90001),
+('101 Pine St', 'New York', 'NY', 10001),
+('202 Oak St', 'Dallas', 'TX', 75201);
+
+
+SELECT CONCAT_WS(Street,City,State,ZIP) AS Address
+FROM Address;
+
+-- 19 Change all occurrences of 'a' to 'e' in the string 'database management'.
+
+SELECT REPLACE('database management','a','e') AS ChangedString;
+
+-- 20 From a 'Products' table, write a query to replace all spaces in product names with underscores.
+
+INSERT INTO Products VALUES
+('7JDFT','AKKK III OO');
+
+UPDATE Products
+SET ProductName = REPLACE(ProductName,' ','-');
```

Comment:
Should replace with underscores, not hyphens

Reply:
okay